File: Atarashii/src/main/java/net/somethingdreadful/MAL/broadcasts/RecordStatusUpdatedReceiver.java
package net.somethingdreadful.MAL.broadcasts;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class RecordStatusUpdatedReceiver extends BroadcastReceiver {
    public static final String RECV_IDENT = "net.somethingdreadful.MAL.broadcasts.RecordStatusUpdatedReceiver";
    private final RecordStatusUpdatedListener callback;

    public RecordStatusUpdatedReceiver(RecordStatusUpdatedListener callback) {
        this.callback = callback;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equals(RECV_IDENT)) {
            if (callback != null) {
                boolean isAnime = (boolean) intent.getSerializableExtra("type");
                callback.onRecordStatusUpdated(isAnime);
            }
        }
    }

    public interface RecordStatusUpdatedListener {
        void onRecordStatusUpdated(boolean isAnime);
    }
}

File: Atarashii/src/main/java/net/somethingdreadful/MAL/broadcasts/RecordStatusUpdatedReceiver.kt
package net.somethingdreadful.MAL.broadcasts

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class RecordStatusUpdatedReceiver(private val callback: RecordStatusUpdatedReceiver.RecordStatusUpdatedListener?) : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == RECV_IDENT) {
            if (callback != null) {
                val isAnime = intent.getSerializableExtra("type") as Boolean
                callback.onRecordStatusUpdated(isAnime)
            }
        }
    }

    interface RecordStatusUpdatedListener {
        fun onRecordStatusUpdated(isAnime: Boolean)
    }

    companion object {
        val RECV_IDENT = "net.somethingdreadful.MAL.broadcasts.RecordStatusUpdatedReceiver"
    }
}